baseclasses/testing/pyRegTest.py

```diff
@@ -263,23 +263,25 @@ def par_add_norm(self, name, values, **kwargs):
     # Private functions
     # *****************
     def assert_allclose(self, actual, reference, name, rtol, atol, full_name=None):
+        """This is basically a wrapper on numpy.testing.assert_allclose with a generated error message"""
         if full_name is None:
             full_name = name
         msg = f"Failed value for: {full_name}"
-        # special case if we're comparing list of strings
-        if isinstance(reference, list) and isinstance(reference[0], str):
-            self.assert_equal(actual, reference, name)
-        else:
-            numpy.testing.assert_allclose(actual, reference, rtol=rtol, atol=atol, err_msg=msg)
+        numpy.testing.assert_allclose(actual, reference, rtol=rtol, atol=atol, err_msg=msg)
```

Comment:
So now is on the user to call the right function, instead of figuring it out with an if check, right?

Reply:
No, this is still detected automatically, but it's done in `_add_values` instead. I also switched from checking dtype to using numpy which is more reliable.